src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -637,6 +637,75 @@ public function get_unsupported_exception() {
 		return $this->unsupported_exception;
 	}
 
+	/**
+	 * Progress through a document pausing on tags matching the provided CSS selector string.
+	 *
+	 * @example
+	 *
+	 *     $processor = WP_HTML_Processor::create_fragment(
+	 *         '<meta charset="utf-8"><title>Example</title><meta property="og:type" content="website"><meta property="og:description" content="An example.">'
+	 *     );
+	 *     foreach ( $processor->select_all( 'meta[property^="og:" i]' ) as $_ ) {
+	 *         // Loop is entered twice.
+	 *         var_dump(
+	 *             $processor->get_tag(),                   // string(4) "META"
+	 *             $processor->get_attribute( 'property' ), // string(7) "og:type" / string(14) "og:description"
+	 *             $processor->get_attribute( 'content' ),  // string(7) "website" / string(11) "An example."
+	 *         );
+	 *     }
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param string $selector_string Selector string.
+	 * @return Generator<void> A generator pausing on each tag matching the selector.
+	 */
+	public function select_all( $selector_string ): Generator {
```

Comment:
I’m not sure this is necessary because the interface of `select()` would provide the same interface as this, but without the dummy `$_` variable, would it not?

```php

while ( $processor->select( 'meta[property^="og:" i]' ) ) {
	…
}
```

except here we also have a generator and non-true/false return values. is there a reason to add a second almost-identical interface to the main loop?

I ask because I think it’s important that developers not have too many ways to do the same thing, and we’ve already tried to communicate the main interface to the HTML API as being this new "loop" in a `while ( $processor->next_token() )` or `while ( $processor->next_tag() )` construct.

Reply:
I agree with this. We addressed it in dc789e4.